test/suites/regression/nodeclaim_test.go

```diff
@@ -248,19 +247,20 @@ var _ = Describe("NodeClaim", func() {
 		})
 		It("should delete a NodeClaim if it references a NodeClass that isn't Ready", func() {
 			env.ExpectCreated(nodeClass)
-			nodeClass = env.ExpectNodeClassCondition(nodeClass, []status.Condition{
-				{
-					Type:               "Ready",
-					Status:             metav1.ConditionFalse,
-					LastTransitionTime: metav1.Now(),
-					Reason:             "NotReady",
-					Message:            "NodeClass is not ready",
-				},
+			// Give 5 seconds to allow Karpenter to update the NodeClass
+			time.Sleep(5 * time.Second)
```

Comment:
Why do we need to sleep and why can't we create an Eventually to wait until we get into a certain condition? I'm always weary of sleeps like this since they too can be flaky.

Reply:
Sure, I updated to add say if we get a condition struct then, the Karpenter has reconciled against the nodeclass.